### DISCUSSION TITLE
Push Notifications are not coming to real device where as working in simulator

### DISCUSSION THREAD

**Discussion by srinivasaraosaluru** [apache/cordova#418] [Q&A]
Hi Team, 

Greetings of the day! 

 We are facing one very weird error during our Hybrid Application Development, looking for some help from our cordove ionic experts. 

Currently we are trying to send push notifications to Hybrid angular ionic application using SAP Emarsys ominichannel to IOS devices.

All notifications are received while testing application in with demo angular ionic application with real device as well as Simulators.  But same is failing after publishing the ipa from aws devops/jenkins script, there are no build errors, no build warnings also we are using p8 certificate to send these device APN tokens. 

 Any one have any previous similar experience, kindly help with your comments. 

 Thank you in advance.

**Comment by jcesarmobile** [comment]
If you are using certificates, they are different for production and for development, make sure you are using the production push certificate.
Or consider moving to token based communication 
https://developer.apple.com/documentation/usernotifications/setting_up_a_remote_notification_server/establishing_a_token-based_connection_to_apns

**Comment by BigBalli** [comment]
Does Emarsys report anything?
Are you using the same device you used for testing?
Does it work if you send push notifications manually?
There is a lot that can break the chain to successful delivery.
